src/OrchardCore.Modules/OrchardCore.OpenId/Configuration/OpenIdClientConfiguration.cs

```diff
@@ -118,11 +118,13 @@ public void Configure(string name, OpenIdConnectOptions options)
         private async Task<OpenIdClientSettings> GetClientSettingsAsync()
         {
             var settings = await _clientService.GetSettingsAsync();
-            if ((await _clientService.ValidateSettingsAsync(settings)).Any(result => result != ValidationResult.Success))
+            var result = await _clientService.ValidateSettingsAsync(settings);
+            if (result.Any(result => result != ValidationResult.Success))
             {
                 if (_shellSettings.State == TenantState.Running)
                 {
-                    _logger.LogWarning("The OpenID Connect module is not correctly configured.");
+                    var errors = result.Where(x => x != ValidationResult.Success).Select(x => x.ErrorMessage);
```

Comment:
These all need to be wrapped in a 
`if (_logger.IsEnabled(LogLevel.Warning))`

Reply:
@deanmarcussen ,  Yes you are right, but I found a problem while updating the code: why should null be returned when a warning occurs? If it is just a warning, it should not affect the function.

``` C#
            var settings = await _clientService.GetSettingsAsync();
            if (_logger.IsEnabled(LogLevel.Warning) &&
                (await _clientService.ValidateSettingsAsync(settings)).Any(result => result != ValidationResult.Success))
            {
                if (_shellSettings.State == TenantState.Running)
                {
                    _logger.LogWarning("The OpenID Connect module is not correctly configured.");
                }

                return null;
            }

            return settings;
```